### ISSUE TITLE
Uploaded file is empty

### ISSUE THREAD

**Issue by RidaTazi** [fabric8io/kubernetes-client#5043]
### Describe the bug

When a file is uploaded, the file is created on the pod, but has no content.

### Fabric8 Kubernetes Client version

6.4.1

### Steps to reproduce

```
    public void uploadFileToPod(KubernetesClient client){
        PrintWriter writer;
        try {
            writer = new PrintWriter("fileToUpload.txt", "UTF-8");
        } catch (FileNotFoundException | UnsupportedEncodingException e) {
            throw new RuntimeException(e);
        }
        writer.println("The first line");
        writer.println("The second line");
        writer.flush();
        writer.close();
        client.pods()
                .withName(podName)
                .file("/dest/fileUploaded.txt")
                .upload(Paths.get("fileToUpload.txt"));
    }
```

I create the file then I upload it, in my local machine the content of the file fileToUpload.txt is present, on the pod fileUploaded.txt is present but it's empty

### Expected behavior

a cat /dest/fileUploaded.txt should print:
The first line
The second line

### Runtime

Kubernetes (vanilla)

### Kubernetes API Server version

1.25.3@latest

### Environment

Windows

### Fabric8 Kubernetes Client Logs

```shell
No logs, the operation succeeds.
```

**Comment by shawkins** [comment]
There is a protocol bug in kubernetes that makes this possible - see https://github.com/kubernetes/kubernetes/issues/89899 and https://github.com/fabric8io/kubernetes-client/pull/4968 

We are now guarding against it in 6.6 with #4968 - the upload boolean value will be false if the upload does not succeed.

**Comment by RidaTazi** [comment]
Okay, is there any workaround to upload files in the current versions?
Apart from putting the file in a dir and uploading the dir ?

**Comment by shawkins** [comment]
> Okay, is there any workaround to upload files in the current versions?

There aren't any great workarounds that I could easily identify.  The least hackish solution so far was just to verify the upload, which you can also do  - exec a "wc -c < filename" and check the size is as expected.

**Comment by manusa** [comment]
This was addressed in [v6.6.0](https://github.com/fabric8io/kubernetes-client/releases/tag/v6.6.0) which was recently released. You will now get an Exception if the uploaded file is incomplete.

**Comment by manusa** [comment]
Closing now, please reopen if you are still encountering issues.
